lib/realtime/tenants/scheduled_message_cleanup.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Realtime.Tenants.ScheduledMessageCleanup do
+  @moduledoc """
+  Scheduled tasks for the Tenants.
+  """
+
+  use GenServer
+  require Logger
+
+  import Ecto.Query
+  import Realtime.Helpers, only: [log_error: 2]
+
+  alias Realtime.Api.Tenant
+  alias Realtime.Database
+  alias Realtime.Messages
+  alias Realtime.Nodes
+  alias Realtime.Repo
+
+  defstruct timer: nil, region: nil, chunks: nil
+
+  def start_link(_args) do
+    timer = Application.get_env(:realtime, :schedule_clean, :timer.hours(5))
+    region = Application.get_env(:realtime, :region)
+    chunks = Application.get_env(:realtime, :chunks, 10)
+    state = %__MODULE__{timer: timer, region: region, chunks: chunks}
+    GenServer.start_link(__MODULE__, state, name: __MODULE__)
+  end
+
+  @impl true
+  def init(%__MODULE__{timer: timer} = state) do
+    Process.send_after(self(), :delete_old_messages, timer)
+    Logger.info("ScheduledMessageCleanup started")
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_info(:delete_old_messages, state) do
+    %{region: region, chunks: chunks, timer: timer} = state
+    regions = Nodes.region_to_tenant_regions(region)
+    region_nodes = Nodes.region_nodes(region)
+
+    Realtime.Repo.transaction(fn ->
+      base = from(t in Tenant, select: t)
```

Comment:
yes it will check all in a loop but we do need to keep it like this. for users that do not have a lot of activity they would potentially never be cleaned up right?

Reply:
yeah, but my main concern is that every node will go through the entire table every time

Reply:
to be tackled in future PR by adding a `last_active_at` which will be set on Realtime.Tenants.Connect process shutdown